File: sdi3-14.Cli-REST-CSharp/sdi3-14.Cli-REST-CSharp/menu/MenuItemExecutor.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace sdi3_14.Cli_REST_CSharp.menu
{
    class MenuItemExecutor
    {
        private IList<MenuItem> items;

        public MenuItemExecutor(IList<MenuItem> items)
        {
            this.items = items;
        }

        public void execute()
        {
            while (true)
            {
                // print the options
                System.Console.WriteLine("\nSeleccione una: \n");

                System.Console.WriteLine("0 - Salir");
                System.Console.WriteLine();

                int counter = 0;
                foreach (MenuItem item in items)
                {
                    System.Console.WriteLine(counter + 1 + " - " + item.GetMenuName());
                    counter++;
                }
                // get the user option
                int userOption = getIntFromUser("Opcion");

                if (userOption > items.Count() || userOption < 0)
                {
                    System.Console.WriteLine("Opcion marcada incorrecta");
                }
                else if (userOption == 0)
                {
                    System.Console.WriteLine("Saliendo");
                    break;
                }
                else
                {
                    this.items = items.ElementAt(userOption - 1).Execute();
                    // this.items.add(new ExitMenuItem());
                }
            }
        }

        private int getIntFromUser(String message)
        {
            
            Int32 userInput = -1;

            do {
                System.Console.Write(message + ": ");
                String result = Console.ReadLine();

                if(Int32.TryParse(result, out userInput))
                    return userInput;

                System.Console.WriteLine("Formato de numero incorrecto");
            } while (true);

        }

    }
}

File: sdi3-14.Cli-REST/src/uo/sdi/clients/rest/menu/MenuItemExecutor.java
package uo.sdi.clients.rest.menu;

import java.util.List;

import alb.util.console.Console;

public class MenuItemExecutor {

	private List<MenuItem> items;

	public MenuItemExecutor(List<MenuItem> items) {
		this.items = items;
	}

	public void execute() {
		while (true) {
			// print the options
			Console.println("\nSeleccione una: \n");

			Console.println("0 - Salir");
			Console.println();
			for (int i = 0; i < items.size(); i++) {
				Console.println(i + 1 + " - " + items.get(i).getMenuName());
			}
			// get the user option
			Integer userOption = Console.readInt("Opcion");
			if (userOption == null) {
				userOption = -1;
			}

			if (userOption > items.size() || userOption < 0) {
				Console.println("Opcion marcada incorrecta");
			} else if (userOption == 0) {
				Console.println("Saliendo");
				break;
			} else {
				this.items = items.get(userOption - 1).execute();
			}
		}
	}
}
